### PR TITLE
LG-2342: Description props accept `React.ReactNode` instead of strings

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-2342](https://jira.mongodb.org/browse/LG-2342)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/select/src/Select/Select.types.ts [hunk 1]
```diff
@@ -41,6 +43,11 @@ export interface BaseSelectProps
    * HTML `id` property used to allow Javascript, form, or label to reference the input.
    */
   id?: string;
+
+  /**
+   * Determines whether or not the component appears in dark theme.
+   * @default false
+   */
   darkMode?: boolean;
```
Comment (shaneeza) [inline packages/select/src/Select/Select.types.ts:51]: not blocking: if you want you can extend `DarkModeProps` here

File: packages/combobox/README.md [hunk 2]
```diff
@@ -87,7 +87,7 @@ import { Combobox, ComboboxOption } from '@leafygreen-ui/combobox';
 | `overflow`               | `'expand-y'`, `'expand-x'`, `'scroll-x'`      | Defines the overflow behavior of a multiselect combobox. **expand-y**: Combobox has fixed width, and additional selections will cause the element to grow in the block direction. **expand-x**: Combobox has fixed height, and additional selections will cause the elemenet to grow in the inline direction. **scroll-x**: Combobox has fixed height and width, and additional selections will cause the element to be scrollable in the x (horizontal) direction. | `'expand-y'`             |
 | `label`                  | `string`                                      | An accessible label for the input, rendered in a `<label>` to the DOM                                                                                                                                                                                                                                                                                                                                                                                               |                          |
```
Comment (shaneeza) [inline packages/combobox/README.md:88]: Should the same be done for `label`?

Comment (bruugey) [reply packages/combobox/README.md:88]: Potentially - this ticket only asked for Description. But WDYT?

Comment (shaneeza) [reply packages/combobox/README.md:88]: Let's create a ticket for the labels to track it. And if it's not too much, we can get it done in this PR

**Review by shaneeza — APPROVED** [review]

File: packages/combobox/README.md (hunk 2 above)
Comment (bruugey) [reply packages/combobox/README.md:88]: Will create a ticket - I think this will actually bea good thing to tackle as a part of CCP.

**Pull request merged**